src/main/java/vic/cinema/dao/ShoppingCartDao.java

```diff
@@ -0,0 +1,13 @@
+package vic.cinema.dao;
+
+import java.util.Optional;
+import vic.cinema.model.ShoppingCart;
+import vic.cinema.model.User;
+
+public interface ShoppingCartDao {
+    ShoppingCart add(ShoppingCart shoppingCart);
+
+    Optional<ShoppingCart> getByUser(User user);
```

Comment:
according to our app logic we're not obligated to use optional here, but it's up to you

Reply:
Done